src/odemis/gui/comp/overlay/base.py

```diff
@@ -1079,6 +1106,444 @@ def _on_motion(self, evt):
+                ),
+                EdgeBoundingBox(
+                    index=4,
+                    l=(self.v_point4.x + self.v_point1.x) / 2 - self.hover_margin,
+                    r=(self.v_point4.x + self.v_point1.x) / 2 + self.hover_margin,
+                    t=(self.v_point4.y + self.v_point1.y) / 2 - self.hover_margin,
+                    b=(self.v_point4.y + self.v_point1.y) / 2 + self.hover_margin,
+                ),
+            ]
+
+            vertices = [
+                EdgeBoundingBox(
+                    index=1,
+                    l=self.v_point1.x - self.hover_margin,
+                    r=self.v_point1.x + self.hover_margin,
+                    t=self.v_point1.y - self.hover_margin,
+                    b=self.v_point1.y + self.hover_margin,
+                ),
+                EdgeBoundingBox(
+                    index=2,
+                    l=self.v_point2.x - self.hover_margin,
+                    r=self.v_point2.x + self.hover_margin,
+                    t=self.v_point2.y - self.hover_margin,
+                    b=self.v_point2.y + self.hover_margin,
+                ),
+                EdgeBoundingBox(
+                    index=3,
+                    l=self.v_point3.x - self.hover_margin,
+                    r=self.v_point3.x + self.hover_margin,
+                    t=self.v_point3.y - self.hover_margin,
+                    b=self.v_point3.y + self.hover_margin,
+                ),
+                EdgeBoundingBox(
+                    index=4,
+                    l=self.v_point4.x - self.hover_margin,
+                    r=self.v_point4.x + self.hover_margin,
+                    t=self.v_point4.y - self.hover_margin,
+                    b=self.v_point4.y + self.hover_margin,
+                ),
+            ]
+
+            self.v_edges.update(
+                {
+                    gui.HOVER_ROTATION: rotation,
+                    gui.HOVER_LINE: midpoints,
+                    gui.HOVER_EDGE: vertices
+                }
+            )
+
+    def update_hover_direction(self, idx):
+        if idx == 1:
+            point1 = self.v_point1
+            point2 = self.v_point2
+        elif idx == 2:
+            point1 = self.v_point2
+            point2 = self.v_point3
+        elif idx == 3:
+            point1 = self.v_point3
+            point2 = self.v_point4
+        elif idx == 4:
+            point1 = self.v_point4
+            point2 = self.v_point1
+        dx = abs(point1.x - point2.x)
+        dy = abs(point1.y - point2.y)
+        return gui.HOVER_DIRECTION_EW if dy > dx else gui.HOVER_DIRECTION_NS
+
+    def get_hover(self, vpos) -> Tuple[int, Optional[int]]:
+        """
+        Check if the given position is on/near a selection edge or inside the selection.
+
+        :return: (tuple) the first element is False if not hovering, or the type of hover.
+            The second element is the index of the v_point if the edit mode is EDIT_MODE_POINT.
```

Comment:
When returning multiple values, it's nicer to give a name to each variable, and explain independently.
```suggestion
        :return:
           hover_type (HOVER_*): the type/location of hover, or HOVER_NONE if not hovering, 
           edge: the index of the v_point involved, if the edit mode is EDIT_MODE_POINT. Otherwise None.
```